### PR TITLE
Use Build.current().minCompat methods instead of Version when building strings

### PR DESCRIPTION
**Pull request by williamrandolph**
We can remove uses of the Version class by getting a string representation of the version from the `Build.current()` object.

### CODE REVIEW

**Review by rjernst — APPROVED** [review]
This is a good start to migrating away from current. More will be needed (in followups), specifically to minCompatibilityVersion. While the string exception messages are changed here, the conditions right above these messages are left using Version. I think we will want a special method specifically for checking compatibility. But this PR as is seems fine as a first step.

File: server/src/test/java/org/elasticsearch/env/NodeMetadataTests.java [hunk 1]
```diff
@@ -137,9 +138,9 @@ public void testDoesNotUpgradeAncientVersion() {
                 startsWith("cannot upgrade a node from version ["),
                 endsWith(
                     "] directly to version ["
-                        + Version.CURRENT
+                        + Build.current().version()
                         + "], upgrade to version ["
-                        + Version.CURRENT.minimumCompatibilityVersion()
+                        + Build.current().version()
```
Comment (rjernst) [inline server/src/test/java/org/elasticsearch/env/NodeMetadataTests.java:143]: this should be Build.current().minimumCompatibilityVersion().

File: test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java [hunk 2]
```diff
@@ -311,9 +312,9 @@ protected String getTestReadinessPorts() {
     public static class VersionSensitiveWarningsHandler implements WarningsHandler {
         Set<String> requiredSameVersionClusterWarnings = new HashSet<>();
         Set<String> allowedWarnings = new HashSet<>();
-        final Set<Version> testNodeVersions;
+        final Set<String> testNodeVersions;
```
Comment (ldematte) [inline test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java:315]: :D I made the same change! Good :)

**Review by ldematte — APPROVED** [review]
LGTM bar Ryan comment on version -> minWireCompatVersion

**Pull request merged**